pandas/core/series.py

```diff
@@ -549,17 +552,13 @@ def _set_axis(self, axis: int, labels, fastpath: bool = False) -> None:
             ):
                 try:
                     labels = DatetimeIndex(labels)
-                    # need to set here because we changed the index
-                    if fastpath:
-                        self._mgr.set_axis(axis, labels)
                 except (tslibs.OutOfBoundsDatetime, ValueError):
                     # labels may exceeds datetime bounds,
                     # or not be a DatetimeIndex
                     pass
 
-        if not fastpath:
-            # The ensure_index call above ensures we have an Index object
-            self._mgr.set_axis(axis, labels)
+        # The ensure_index call above ensures we have an Index object
+        self._mgr.set_axis(axis, labels)
```

Comment:
what about case with fastpath=True and the DatetimeIndex call raising?

Reply:
`_set_axis` never gets called when `fastpath=True` (in the current PR, to be clear)

Reply:
> (in the current PR, to be clear)

OK.  is there a danger that it could be reached by a user?  i know the places where we use fastpath=True internally are pretty strict, but it is unfortunately exposed in the API

Reply:
On master, if you specify `fastpath=True`, it is already required that `index` is an actual Index object (otherwise you would get an error about not having an `_is_all_dates` attribute). So I think it is not that likely that this change would cause a problem.